### PR TITLE
Update to latest rustc

### PR DESCRIPTION
**Pull request by GBGamer**
* change from `Vec::from_elem(x, y)` to `repeat(y).take(x).collect()`
* change from `task_rng` to `thread_rng`
* change from `trim_chars` to `trim_matches`

### CODE REVIEW

**Comment by GBGamer** [comment]
I don't know why travis isn't attempting to build this branch

**Comment by GBGamer** [comment]
I just reset --soft, recommitted, and force pushed to attempt to get travis to build. I guess it worked.

File: examples/staging/macros/dry.rs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #![feature(macro_rules)]
+use std::iter::repeat;
```
Comment (steveklabnik) [inline examples/staging/macros/dry.rs]: better style is to `use std::iter` and then use `iter::repeat`

**Comment by steveklabnik** [comment]
r=me after updating to the better style. Thanks so much!

**Comment by GBGamer** [comment]
@steveklabnik done!

**Comment by steveklabnik** [comment]
The build is failing for an unrelated reson, so I'm merging. Thank you!

**Pull request merged**